src/cmd-buildextend-metal

```diff
@@ -208,23 +208,15 @@ echo "Estimating disk size..."
 # and doing so has apparently negative performance implications.
 /usr/lib/coreos-assembler/estimate-commit-disk-size ${BLKSIZE:+--blksize ${BLKSIZE}} --repo "$ostree_repo" "$commit" --add-percent 35 > "$PWD/tmp/ostree-size.json"
 rootfs_size="$(jq '."estimate-mb".final' "$PWD/tmp/ostree-size.json")"
-# extra size is the non-ostree partitions, see create_disk.sh
-image_size="$(( rootfs_size + 513 ))M"
-echo "Disk size estimated to ${image_size}"
-
-# For bare metal and dasd images, we use the estimated image size. For IaaS/virt, we get it from
-# image.yaml because we want a "default" disk size that has some free space.
-case "${image_type}" in
-    metal*|dasd)
-        # Unset the root size, which will inherit from the image size
-        rootfs_size=0
-        ;;
-    qemu)
-        image_size="$(jq -r ".size" < "${image_json}")G"
-        rootfs_size="${rootfs_size}M"
-        ;;
-    *) fatal "unreachable image_type ${image_type}";;
-esac
+# The minimum size of a disk image we'll need will be the rootfs_size
+# estimate plus the size of the non-root partitions. We'll use this
+# size for the metal/dasd images, but for the IaaS/virt image we'll use
+# the size set in the configs since some of them have minimum sizes that
+# the platforms require and we ant a "default" disk size that has some
+# free space.
+metal_image_size="$(( rootfs_size + 513 ))M"
+cloud_image_size="$(jq -r ".size" < "${image_json}")G"
+echo "Disk sizes: metal: ${metal_image_size} (estimated), cloud: ${cloud_image_size}"
```

Comment:
Minor: this is a kinda confusing log message since we're only building one of the two.

Reply:
yeah. I kind of do want this information printed out, though. ideas?

The truth is that now we will be using both of these values when a `qemu` image is being built even though that's an implementation detail.

Reply:
Yeah, but we're not using `cloud_image_size` at all in the non-`qemu` path. So we could actually not even calculate it at all.

E.g. could do something like:

```sh
initial_image_size="$(( rootfs_size + 513 ))M"
echo "Initial image size: ${initial_image_size}
...
    # Now that we've created the image let's resize it to the desired
    # size for qemu (and all images based on it).
    if [[ $image_type == qemu ]]; then
        cloud_image_size="$(jq -r ".size" < "${image_json}")G"
        qemu-img resize "${path}.tmp" "${cloud_image_size}"
        echo "Resized image to: ${cloud_image_size}"
    fi
```

Reply:
I think it may be useful to see the later patches where this is used too (opened https://github.com/coreos/coreos-assembler/pull/3695 - see the `support dynamically setting the disk size` commit). In the osbuild path it's easier if we just pass both pieces of information in and since it's not expensive to read a value from a json file I just decided to do that.

Reply:
Sounds good. Feel free to merge this one and we can chat in #3695.

Reply:
> Sounds good. Feel free to merge this one and we can chat in #3695.

done